SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialPackageMapClient.cpp

```diff
@@ -832,3 +865,65 @@ void FSpatialNetGUIDCache::RegisterObjectRef(FNetworkGUID NetGUID, const FUnreal
 	NetGUIDToUnrealObjectRef.Emplace(NetGUID, RemappedObjectRef);
 	UnrealObjectRefToNetGUID.Emplace(RemappedObjectRef, NetGUID);
 }
+
+void FSpatialNetGUIDCache::SlowCheckMapsConsistency(const UObject* Object, const FString& CheckCallsite) const
+{
+#if DO_GUARD_SLOW
+	const FNetworkGUID ObjectNetGUID = GetNetGUID(Object);
+	const FUnrealObjectRef ObjectRef = GetUnrealObjectRefFromNetGUID(ObjectNetGUID);
+	const FNetworkGUID ObjRefNetGUID = UnrealObjectRefToNetGUID.FindRef(ObjectRef);
+	;
+
+	if (ObjectNetGUID != ObjRefNetGUID)
+	{
+		UE_LOG(LogSpatialPackageMap, Error,
+			   TEXT("NetGUID inconsistency between SpatialNetGuidCache and native's NetGuidCache. Object: %s, ObjectNetGUID: %u, "
+					"ObjRefNetGUID: %u"),
+			   *GetNameSafe(Object), ObjectNetGUID.Value, ObjRefNetGUID.Value);
+	}
+
+	if (!IsValid(Object))
+	{
+		if (ObjectNetGUID.IsValid())
+		{
+			UE_LOG(LogSpatialPackageMap, Error,
```

Comment:
maybe this could even be an ensureAlwaysMsgf so we get a callstack with it.
alternative there is a dump stack trace function somewhere

Reply:
ensureAlwaysMsgf is probably better, have changed